R/dock_from_renv.R

```diff
@@ -0,0 +1,90 @@
+#' Create a Dockerfile from an `renv.lock` file
+#'
+#' @param lockfile Path to an `renv.lock` file.
+#' @param distro One of "focal", "bionic", "xenial", "centos7", or "centos8". See available distributions at https://hub.docker.com/r/rstudio/r-base/.
+#' @param primary_repo Default "CRAN". Indicates which repository in the `renv.lock` file to install the renv package from.
+#' @param out_dir The directory where to write the `Dockerfile`. When `NULL` no file is written. `NULL` by default.
+#' @param use_rspm Utilize the Public RStudio Package Manager for Linux binaries. Default `TRUE`.
+#'
+#' @return A R6 object of class `Dockerfile`.
+#' @details
+#'
+#' System requirements for packages are provided through RStudio Package Manager via the {pak} package. The install commands provided from pak are added as `RUN` directives within the `Dockerfile`.
+#'
+#' The R version is taken from the `renv.lock` file. Packages are installed using `renv::restore()` which ensures that the proper package version and source is used when installed.
+#'
+#' By default, `dock_from_renv()` will utilize the [public RStudio Package Manager](https://packagemanager.rstudio.com/) (PPM) to provide linux binaries. This dramatically improves install time. This is done when `use_rspm = TRUE`.  To do so, a new file `renv.lock.dock` will be created so as to not overwrite your existing `renv.lock` file. This new `renv.lock.dock` file is used to build the Docker image with PPM binaries.
+#'
+#' \donttest{
+#' dock <- dock_from_renv("renv.lock", distro = "xenial", out_dir = getwd())
+#' dock
+#' }
+#' @export
+dock_from_renv <- function(lockfile = "renv.lock", distro = "xenial", primary_repo = "CRAN",
+                           out_dir = NULL,
+                           use_rspm = TRUE) {
+
+  lock <- renv:::lockfile(lockfile)
```

Comment:
Can you refactor his with `getFromNamespace("lockfile", "renv")(lockfile)`, so that it passes the CRAN? 

Thanks

Reply:
`lockfile$version()` might be a versioning issue. Will adjust to use `lock$data()$R$Version`
https://github.com/rstudio/renv/blob/f40560d6f94a06c363f6a082d0ea1b187f3bde16/R/lockfile-api.R#L32